### DISCUSSION TITLE
Gameplay and audio slowed down when using pipewire JACK audio output.

### DISCUSSION THREAD

**Discussion by LeadSeason** [ppy/osu#23085] [Q&A]
### Type

Game behaviour

### Bug description

Gameplay slow down is only caused by pipewire JACK audio server

## Additional version info

```
 $ pacman -Ss pipewire | grep "\[installed\]"
cachyos-v3/libpipewire 1:0.3.67-1.1 [installed]
cachyos-v3/libwireplumber 0.4.14-1.1 [installed]
cachyos-v3/pipewire 1:0.3.67-1.1 [installed]
cachyos-v3/pipewire-alsa 1:0.3.67-1.1 [installed]
cachyos-v3/pipewire-audio 1:0.3.67-1.1 [installed]
cachyos-v3/pipewire-jack 1:0.3.67-1.1 [installed]
cachyos-v3/pipewire-pulse 1:0.3.67-1.1 [installed]
cachyos-v3/wireplumber 0.4.14-1.1 [installed]
multilib/lib32-libpipewire 1:0.3.67-1 [installed]
multilib/lib32-pipewire 1:0.3.67-1 [installed]

 $ yay -Ss osu-lazer | grep "\(Installed\)"
aur/osu-lazer-bin 2023.326.1-1 (+52 3.05) (Installed)
```

### Screenshots or videos

https://user-images.githubusercontent.com/60569772/229350517-a859c694-27d5-4cf8-8803-e49bfd3c22a1.mp4

### Version

2023.326.1-lazer

### Logs

[network.log](https://github.com/ppy/osu/files/11132059/network.log)
[runtime.log](https://github.com/ppy/osu/files/11132060/runtime.log)
[performance.log](https://github.com/ppy/osu/files/11132061/performance.log)
[database.log](https://github.com/ppy/osu/files/11132062/database.log)
[input.log](https://github.com/ppy/osu/files/11132063/input.log)

**Comment by Axilotl17** [comment]
related to #22425 maybe?

**Comment by LeadSeason** [comment]
After further experimenting here are my conclusions.
when pipewire is configure to use only one sample rate everything works as expected, but when multiple sample rates are allowed weird behavior happens.
my pipewire.conf has
```conf
context.properties = {
    default.clock.rate = 384000
    default.clock.allowed-rates = [ 44100 48000 88200 96000 176400 192000 352800 384000 ]
}
```
## conf contains
```conf
default.clock.rate = 384000
default.clock.allowed-rates = [ 44100 48000 88200 96000 176400 192000 352800 384000 ]
```
When osu is opened with pipewire set, sample rate is set to 44100Hz.
when audio device is changed from pipewire -> jack audio and gameplay is slowed down.

When osu is opened with jack set, sample rate is set to 384000Hz.
when audio device is changed from jack -> pipewire. nothing happens.

## conf contains
```conf
default.clock.rate = 44100
default.clock.allowed-rates = [ 44100 48000 88200 96000 176400 192000 352800 384000 ]
```
when osu will work normally with no audio problems unless a audio file of 48000Hz sample rate or higher is played before opening osu and switching from pipewire -> jack will cause audio and gameplay to be speedup

# Gameplay issues
When audio device is switched from pipewire -> jack audio and gameplay is slowed down this give the player an extreme unfair advantage. Take me, It is difficult for me to do a 200pp play but with this one trick I managed to get a 4000pp play second try. (Please don't ban me 😅)
